packages/builtin/array.pony

```diff
@@ -276,6 +276,14 @@ class Array[A] is Seq[A]
 
     this
 
+  fun add(that: this->Array[A]): Array[this->A!] =>
```

Comment:
Can the return type be made an `iso^` for easy flexibility on the caller side (I think the signature you mentioned in #660 did this)?

Also, whatever signature you come up with, you'll want to make sure it is compatible enough with `String.add` to allow you to call `ByteSeq.add` as you mention in #660.